File: api/internal/gateway/admin/v1/request/administrator.go
package request

type CreateAdministratorRequest struct {
	Lastname      string `json:"lastname" validate:"required,max=16"`               // 姓
	Firstname     string `json:"firstname" validate:"required,max=16"`              // 名
	LastnameKana  string `json:"lastnameKana" validate:"required,max=32,hiragana"`  // 姓(かな)
	FirstnameKana string `json:"firstnameKana" validate:"required,max=32,hiragana"` // 名(かな)
	Email         string `json:"email" validate:"required,email"`                   // メールアドレス
	PhoneNumber   string `json:"phoneNumber" validate:"required,e164"`              // 電話番号
}

type UpdateAdministratorRequest struct {
	Lastname      string `json:"lastname" validate:"required,max=16"`               // 姓
	Firstname     string `json:"firstname" validate:"required,max=16"`              // 名
	LastnameKana  string `json:"lastnameKana" validate:"required,max=32,hiragana"`  // 姓(かな)
	FirstnameKana string `json:"firstnameKana" validate:"required,max=32,hiragana"` // 名(かな)
	PhoneNumber   string `json:"phoneNumber" validate:"required,e164"`              // 電話番号
}

type UpdateAdministratorEmailRequest struct {
	Email string `json:"email" validate:"required,email"` // メールアドレス
}

File: web/admin/src/types/validations/administrator.ts
import type { ValidationArgs } from '@vuelidate/core'
import { email, kana, maxLength, required, tel } from '~/lib/validations'

export const CreateAdministratorValidationRules: ValidationArgs = {
  lastname: { required, maxLength: maxLength(16) },
  firstname: { required, maxLength: maxLength(16) },
  lastnameKana: { required, maxLength: maxLength(32), kana },
  firstnameKana: { required, maxLength: maxLength(32), kana },
  email: { required, email },
  phoneNumber: { required, tel },
}

export const UpdateAdministratorValidationRules: ValidationArgs = {
  lastname: { required, maxLength: maxLength(16) },
  firstname: { required, maxLength: maxLength(16) },
  lastnameKana: { required, maxLength: maxLength(32), kana },
  firstnameKana: { required, maxLength: maxLength(32), kana },
  phoneNumber: { required, tel },
}
